pkg/batch/executor_test.go

```diff
@@ -0,0 +1,85 @@
+package batch_test
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/treeverse/lakefs/pkg/batch"
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+func TestExecutor_BatchFor(t *testing.T) {
+	// Setup executor
+	exec := batch.NewExecutor(logging.Dummy())
+	go exec.Run(context.Background())
+	// Prove the executor does not violate read-after-write consistency.
+	// First, let's define read-after-write consistency:
+	// 	Any read that started after a successful write has returned, must return the updated value.
+	// to test this, let's simulate the following scenario:
+	// 1. reader (r1) starts (Current version: v0)
+	// 2. writer (w1) writes v1
+	// 3. writer (w1) returns (Current version: v1)
+	// 4. reader (r2) starts
+	// 5. both readers (r1,r2) return with v1 as their response.
+	var db = sync.Map{}
+	db.Store("v", "v0")
+
+	read1Done := make(chan bool)
+	write1Done := make(chan bool)
+	read2Done := make(chan bool)
+
+	// we pass a custom delay func that ensures we make the write only after
+	//  reader1 started
+	waitWrite := make(chan bool)
+	delays := int32(0)
+	delayFn := func(dur time.Duration) {
+		delaysDone := atomic.AddInt32(&delays, 1)
+		if delaysDone == 1 {
+			close(waitWrite)
```

Comment:
Note that the write can occur before https://github.com/treeverse/lakeFS/pull/1618/files#diff-c9e7aae146c0798d32ade9be6fa5013612246e323b7a2796dbcdc83a0151c607R82 ever happens (because the scheduler is evil).  I think you may need to wait on *another* channel here, that writer1 will close after it _does_ write.